x/stake/keeper/validator.go

```diff
@@ -328,38 +343,48 @@ func (k Keeper) UpdateBondedValidators(ctx sdk.Context,
 			}
 		}
 
-		// if not previously a validator (and unrevoked),
-		// kick the cliff validator / bond this new validator
-		if validator.Status() != sdk.Bonded && !validator.Revoked {
-			kickCliffValidator = true
-
-			validator = k.bondValidator(ctx, validator)
-			if bytes.Equal(ownerAddr, affectedValidator.Owner) {
-				updatedVal = validator
-			}
-		}
-
+		// increment bondedValidatorsCount / get the validator to bond
 		if !validator.Revoked {
-			bondedValidatorsCount++
-		} else {
-			if validator.Status() == sdk.Bonded {
-				panic(fmt.Sprintf("revoked validator cannot be bonded, address: %v\n", ownerAddr))
+			if validator.Status() != sdk.Bonded {
+				validatorToBond = validator
 			}
+			bondedValidatorsCount++
+
+			// sanity check
+		} else if validator.Status() == sdk.Bonded {
+			panic(fmt.Sprintf("revoked validator cannot be bonded, address: %v\n", ownerAddr))
 		}
 
 		iterator.Next()
 	}
 	iterator.Close()
 
-	// perform the actual kicks
-	if oldCliffValidatorAddr != nil && kickCliffValidator {
-		validator, found := k.GetValidator(ctx, oldCliffValidatorAddr)
-		if !found {
-			panic(fmt.Sprintf("validator record not found for address: %v\n", oldCliffValidatorAddr))
-		}
-		k.unbondValidator(ctx, validator)
+	// clear or set the cliff validator
+	if bondedValidatorsCount == int(maxValidators) {
+		k.setCliffValidator(ctx, validator, k.GetPool(ctx))
+	} else if len(oldCliffValidatorAddr) > 0 {
+		k.clearCliffValidator(ctx)
 	}
 
+	// perform the validator swap for any new validators to add
+	if len(validatorToBond.Owner) > 0 {
```

Comment:
Do you think a convenience method would help? `validatorToBond.HasOwner()`

Reply:
All validators should have owners so that function doesn't really make sense... This is really just supposed to be checking if validatorToBond was set - I added a new bool term to make this clearer